public/app/plugins/datasource/prometheus/migrations/variableMigration.ts

```diff
@@ -0,0 +1,95 @@
+import { PromVariableQuery, PromVariableQueryType as QueryType } from '../types';
+
+const labelNamesRegex = /^label_names\(\)\s*$/;
+const labelValuesRegex = /^label_values\((?:(.+),\s*)?([a-zA-Z_][a-zA-Z0-9_]*)\)\s*$/;
+const metricNamesRegex = /^metrics\((.+)\)\s*$/;
+const queryResultRegex = /^query_result\((.+)\)\s*$/;
+
+export function migrateVariableQueryToEditor(rawQuery: string | PromVariableQuery): PromVariableQuery {
+  // If not string, we assume PromVariableQuery
+  if (typeof rawQuery !== 'string') {
+    return rawQuery;
+  }
+
+  const queryBase = {
+    refId: 'PrometheusDatasource-VariableQuery',
+    exprType: QueryType.LabelNames,
+  };
+
+  const labelNames = rawQuery.match(labelNamesRegex);
+  if (labelNames) {
+    return {
+      ...queryBase,
+      exprType: QueryType.LabelNames,
+    };
+  }
+
+  const labelValues = rawQuery.match(labelValuesRegex);
+  if (labelValues) {
+    if (labelValues[2]) {
```

Comment:
Maybe assign some variables to the result of the rawQueryMatch like `const [totalMatch,label,metric] = rawQuery.match(labelValuesRegex);` for readability instead of using raw index values

Reply:
This is good, although I can't assign them all inline because totalMatch could be null and I'm getting this error 
```
Type 'RegExpMatchArray | null' must have a '[Symbol.iterator]()' method that returns an iterator.
```
I can assign the label and the metric a variable to make this more readable. Will do!